sds/src/scanner/mod.rs

```diff
@@ -1029,4 +1033,77 @@ mod test {
         // treated as a false positive.
         assert_eq!(matches.len(), 1);
     }
+
+    #[test]
+    fn panic_test() {
```

Comment:
as this test does not panic, maybe it can be name with another name, `test_calculate_indices_is_called_with_sorted_start_index` for instance.
```suggestion
    fn test_calculate_indices_is_called_with_sorted_start_index() {
```

Reply:
It did panic when I wrote it originally 😆 
I just forgot to change the name, so thanks for noticing.